libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -910,6 +919,50 @@ GCS_MAVLINK::update(uint32_t max_time_us)
     hal.util->perf_end(_perf_update);    
 }
 
+/*
+  record stats about this link to dataflash
+*/
+void GCS_MAVLINK::log_mavlink_stats()
+{
+    enum LogMessages id;
+    switch(chan) {
```

Comment:
The normal way of arriving at this is to just take `LOG_MAV0_MSG` and add the chan to it. (Which I think if that isn't valid we have a lot of other issues elsewhere in the MAVLink stuff). Easy to add static asserts if you want to be paranoid, but it would be faster then the switch stuff.

Reply:
Moved to logging chan directly.